File: src/main/java/tax/bilibili/nineblog/NineBlogApplication.java
package tax.bilibili.nineblog;

import org.mybatis.spring.annotation.MapperScan;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

/**
 * NineBlogApplication
 *
 * @author sheip9
 * @since  2023/10/22 19:20
 */
@SpringBootApplication
@MapperScan("tax.bilibili.nineblog.mapper")
public class NineBlogApplication {

    public static void main(String[] args) {
        SpringApplication.run(NineBlogApplication.class, args);
    }

}

File: server/src/main/java/tax/bilibili/nineblog/NineBlogApplication.kt
package tax.bilibili.nineblog

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.ConfigurationPropertiesScan
import org.springframework.boot.runApplication
import org.springframework.cache.annotation.EnableCaching
import org.springframework.context.annotation.ComponentScan

@EnableCaching
@ConfigurationPropertiesScan
@SpringBootApplication
@ComponentScan(basePackages = ["tax.bilibili.nineblog.application"])
open class NineBlogApplication

@SpringBootApplication
@ComponentScan(basePackages = ["tax.bilibili.nineblog.initialize"])
open class NineBlogInitializeApplication

fun main(args: Array<String>) {
    //TODO: 判断程序配置是否初始化
    if (true) {
        runApplication<NineBlogApplication>(*args)
    } else {
        runApplication<NineBlogInitializeApplication>(*args)
    }
}

